### PR TITLE
Add KubeDNS template

### PR DESCRIPTION
**Pull request by imikushin**
Add KubeDNS template

### CODE REVIEW

File: kubernetes-templates/kube-dns/0/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# KubeDNS
+
+Kubernetes offers a DNS cluster addon, which most of the supported environments
```
Comment (alena1108) [inline kubernetes-templates/kube-dns/0/README.md]: we shouldn't mention cluster add on here. Our services won't be enabled using the add on manager.

File: kubernetes-templates/kube-dns/0/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+enable by default.  We use [SkyDNS](https://github.com/skynetservices/skydns)
+as the DNS server, with some custom logic to slave it to the kubernetes API
```
Comment (alena1108) [inline kubernetes-templates/kube-dns/0/README.md]: could you elaborate on the custom logic to slave to k8s api?

File: kubernetes-templates/kube-dns/0/kubedns-rc.yaml [hunk 3]
```diff
@@ -0,0 +1,111 @@
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: kube-dns-v20
+  namespace: kube-system
```
Comment (alena1108) [inline kubernetes-templates/kube-dns/0/kubedns-rc.yaml]: would kube-system namespace present by default?

File: kubernetes-templates/kube-dns/0/README.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# KubeDNS
```
Comment (alena1108) [inline kubernetes-templates/kube-dns/0/README.md]: we should mention that dns ip address is reserved to 10.43.0.10

File: kubernetes-templates/kube-dns/0/README.md (hunk 1 above)
Comment (imikushin) [inline kubernetes-templates/kube-dns/0/README.md]: will fix

File: kubernetes-templates/kube-dns/0/kubedns-rc.yaml (hunk 3 above)
Comment (imikushin) [inline kubernetes-templates/kube-dns/0/kubedns-rc.yaml]: it is

File: kubernetes-templates/kube-dns/0/README.md (hunk 4 above)
Comment (imikushin) [inline kubernetes-templates/kube-dns/0/README.md]: will fix

File: kubernetes-templates/kube-dns/0/README.md (hunk 2 above)
Comment (alena1108) [inline kubernetes-templates/kube-dns/0/README.md]: We can skip it. Having a link to skydns is good enough.

**Comment by imikushin** [comment]
@alena1108 I believe I addressed your comments.

**Comment by imikushin** [comment]
rancher/rancher#5948

**Pull request closed without merging**